### PR TITLE
Day One

### PR DESCRIPTION
**Pull request by kyranmccann**

### CODE REVIEW

File: basic_hashtable/b_hashtables.c [hunk 1]
```diff
@@ -94,7 +107,17 @@ void hash_table_insert(BasicHashTable *ht, char *key, char *value)
  ****/
 void hash_table_remove(BasicHashTable *ht, char *key)
 {
+  int index = hash(key, ht -> capacity);
 
+  if (ht -> storage[index] == NULL)
+  {
+    printf("ERROR: That key not found");
+    return NULL;
```
Comment (tgreenhalgh) [inline basic_hashtable/b_hashtables.c]: just `return;` `return NULL;` breaks the compiler as this is a `void` function

File: basic_hashtable/b_hashtables.c [hunk 2]
```diff
@@ -94,7 +107,17 @@ void hash_table_insert(BasicHashTable *ht, char *key, char *value)
...
+  }
+  else
```
Comment (tgreenhalgh) [inline basic_hashtable/b_hashtables.c:117]: redundant `else`
```
  if (ht->storage[index] == NULL)
  {
    printf("ERROR: That key not found");
    return;
  }
  destroy_pair(ht->storage[index]);
```

File: basic_hashtable/b_hashtables.c [hunk 3]
```diff
@@ -94,7 +107,17 @@ void hash_table_insert(BasicHashTable *ht, char *key, char *value)
...
+  {
+    destroy_pair(ht -> storage[index]);
```
Comment (tgreenhalgh) [inline basic_hashtable/b_hashtables.c:119]: after `destroy_pair``ht->storage[index]` still has junk in it and it needs to be cleared out

`ht->storage[index] = NULL;`

File: basic_hashtable/b_hashtables.c [hunk 4]
```diff
@@ -104,6 +127,15 @@ void hash_table_remove(BasicHashTable *ht, char *key)
  ****/
 char *hash_table_retrieve(BasicHashTable *ht, char *key)
 {
+  int index = hash(key, ht -> capacity);
+
+  if (ht -> storage[index] != NULL)
+  {
+    if (strcmp(ht -> storage[index] -> key, key) == 0)
```
Comment (tgreenhalgh) [inline basic_hashtable/b_hashtables.c]: don't need this check - `if (ht -> storage[index] != NULL)` does all we need for this one

File: basic_hashtable/b_hashtables.c [hunk 5]
```diff
@@ -104,6 +127,15 @@ void hash_table_remove(BasicHashTable *ht, char *key)
...
+    {
+      return ht -> storage[index] -> value;
+    }
+  }
   return NULL;
```
Comment (tgreenhalgh) [inline basic_hashtable/b_hashtables.c:138]: while this is what the comments strictly called for, you should probably print something saying unable to find such and such index

File: basic_hashtable/b_hashtables.c [hunk 6]
```diff
@@ -114,7 +146,17 @@ char *hash_table_retrieve(BasicHashTable *ht, char *key)
...
+  if (ht != NULL)
+  {
+    for (int i = 0; i < ht -> capacity; i++)
+    {
+      if (ht -> storage[i] != NULL)
+      {
+        destroy_pair(ht -> storage[i]);
+      }
+    }
+    free(ht);
```
Comment (tgreenhalgh) [inline basic_hashtable/b_hashtables.c:157]: you also need to `free(ht->storage);`

**Review by tgreenhalgh — CHANGES_REQUESTED** [review]
looking pretty good!
please change all your `int index = hash(key, ht->capacity);` to `unsigned int` and a couple other small things and you're all set.

Good work!

**Comment by kyranmccann** [comment]
I think I got everything, @tgreenhalgh. let me know if i didn't

**Pull request merged**